### DISCUSSION TITLE
How to generate html with with class attributes?

### DISCUSSION THREAD

**Discussion by szabgab** [wooorm/markdown-rs#105] [Q&A] [answered]
I am using [Bulma](https://bulma.io/) where that uses several classes.

Basically instead if `h1` I'd like to have `<h1 class="title">` etc.

In order to generate the page I have this code:

```rust
let content = markdown2html(&content)
        .replace("<h1>", "<h1 class=\"title\">")
        .replace("<h2>", "<h2 class=\"title is-4\">")
        .replace("<h3>", "<h3 class=\"title is-5\">");

fn markdown2html(content: &str) -> String {
    markdown::to_html_with_options(
        content,
        &markdown::Options {
            compile: markdown::CompileOptions {
                allow_dangerous_html: true,
                ..markdown::CompileOptions::default()
            },
            ..markdown::Options::gfm()
        },
    )
    .unwrap()
}
```

All those calls to `replace` seem to be a bit silly, but I could not find a way to tell  ` markdown::to_html_with_options` to generate it by itself.

How can I convince the crate to use my tags instead of the default ones?

**Comment by wooorm** [comment, marked as answer]
You can’t and you don’t need to. Bulma, next to supporting all those classes, also supports semantic html:
https://bulma.io/documentation/elements/content/

**Reply by lukasturcani** [reply]
what if one uses tailwind (which only allows styling of direct descendants AFAIK)

**Reply by ChristianMurphy** [reply]
Use tailwind prose https://github.com/tailwindlabs/tailwindcss-typography#readme
